### ISSUE TITLE
[3.0] Proposal: getPurchaseStatus

### ISSUE THREAD

**Issue by Jellyfrog** [thephpleague/omnipay#355]
There should be a standard "get status" method for purchases (and other types too?)
Right now each gateway reinvents the wheel, for example:

omnipay-paypal: `fetchPurchase()` https://github.com/thephpleague/omnipay-paypal/blob/master/src/RestGateway.php#L386

omnipay-stripe: `fetchTransaction()` https://github.com/thephpleague/omnipay-stripe/blob/master/src/Gateway.php#L255

I suggest adding a new (optional?) method: `getPurchaseStatus()`

**Comment by delatbabel** [comment]
Yes I agree with this.  I would name the function getPurchaseStatus(). I would be tempted to have a fixed list of statuses (approved, cancelled, refunded, rejected, error ... etc) and map the gateway specific status to one of our fixed list.

**Comment by Jellyfrog** [comment]
Yes, it should use the responseinterface from here:
https://github.com/thephpleague/omnipay/commit/09bfd2719cdd455be8288bd07e4ca36c2120e474
